great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -85,12 +85,14 @@ def __new__(cls, clsname, bases, attrs):
             # Only particular "DataAssistant" implementations must be registered.
             newclass.data_assistant_type = camel_to_snake(name=clsname)
 
-            from great_expectations.rule_based_profiler.data_assistant.data_assistant_dispatcher import (
-                DataAssistantDispatcher,
-            )
+            alias: Optional[str] = getattr(newclass, "__alias__", None)
+            if alias and not alias.startswith("_"):
+                from great_expectations.rule_based_profiler.data_assistant.data_assistant_dispatcher import (
+                    DataAssistantDispatcher,
+                )
 
-            # noinspection PyTypeChecker
-            DataAssistantDispatcher._register_data_assistant(data_assistant=newclass)
+                # noinspection PyTypeChecker
+                DataAssistantDispatcher._register(name=alias, data_assistant=newclass)
```

Comment:
@TrangPham This is nice! 💙 (hope no IDE errors, right?).  Thanks!

Reply:
No errors within my IDE or the CI pipeline.